src/query/execute_statement.c

```diff
@@ -14809,12 +14809,67 @@ do_replicate_statement (PARSER_CONTEXT * parser, PT_NODE * statement)
     {
       PT_PRINT_VALUE_FUNC saved_func = parser->print_db_value;
       int saved_custom_print = parser->custom_print;
+      PARSER_VARCHAR **host_val;
+      char *sbr_text, *sql_text;
+      int i, n, nth;
+      int sql_len = strlen (parser->context.sql_user_text);
+      int var_len = 0;
+      bool begin_quote = false;
+
+      host_val = (PARSER_VARCHAR **) malloc (sizeof (PARSER_VARCHAR *) * parser->host_var_count);
+      if (host_val == NULL)
+	{
+	  return ER_OUT_OF_VIRTUAL_MEMORY;
+	}
+
+      for (i = 0; i < parser->host_var_count; i++)
+	{
+	  host_val[i] = pt_print_db_value (parser, &parser->host_variables[i]);
+	  var_len += host_val[i]->length;
+	}
+
+      sbr_text = (char *) malloc (sql_len + var_len);
```

Comment:
When allocing memory, I think, you will change memory size to sql_len+var_len+1 for adding a null character into the last character.

Reply:
'?' roles 1 byte memory, so the sql_len + var_len is enough,

Reply:
It seems that the sql_len and var_len values ​​do not contain values ​​for handling null strings in sbr_text.

Reply:
the value of host variable does not need to include a null byte because the value is not the end of the query string.
if the sql is "insert into sbr_t values (?, ?)" and host variables are 'abcd' and 'cdef' each other, the final query string is "insert into sbr_t values ('abcd', 'cdef')". the final query string's length is strlen(sql) + strlen('abcd') + strlen('cdef'). it's enough to include all of host values.

Reply:
OK, I got it.